docs/design/jet/025-jet-job-member-pruning.md

```diff
@@ -0,0 +1,192 @@
+- (IMap) Partition pruning - extract partition condition and assign only required partition to be read by `ReadMapOrCacheP`;
+
+## Goals
+
+The goals of that initiative is corresponding with items enumerated in 'Terminology' section : 
+
+- deploy a job only on members which contain required partitions.
+- prune processors which are not required for job execution - e.g, their input processor doesn't produce data and 
+the processor itself can work only with input.
+- limit number of partitions involved in IMap scan.
+
+Non-goals are:
+
+- support any kind of pruning for streaming jobs. It may be considered to do later, but now it is not a case.
+- support migration-tolerance for member and processor pruning. 
+- support local index scan.
+
+## Technical Design
+
+Let's split member pruning, processor pruning and partition pruning as separate stages, which can complement each other, 
+but on their own, each kind of pruning is still a good optimization.
+
+### Member pruning
+
+Member pruning by its own, may have pretty simple solution. To implement it, we need to support that solution 
+in `ExecutionPlanBuilder` (execution plan creation phase). Previous `ProcessorMetaSupplier#get` contract was a reason 
+for strict assumptions in execution plan creation algorithm, but that has changed.
+
+#### Use cases for member pruning
+
+**Important note**: by default, the basic unit on example pictures will be DAG vertices, but sometimes we will 
+use processors as basic unit instead. We will explicitly declare it.
+
+##### Member pruning for simple map scan
+
+Prune all members except Member 1 for scanning job. We want to highlight this simple case, because we assume that 
+the bigger part of the submitted queries looks like `SELECT * FROM map WHERE ...`. So, if you may hint the optimizer
+with partitionKey, the submitted job could be even local.
+
+```
+- SCAN[map2, partitionKey=1]
+```
+
+##### Member pruning for Scan -> Hash Join in big cluster
+
+Prune members without any connections, so, only Member 1 and Member K left in cluster.
+
+```
+- HASH JOIN
+    - SCAN[map1, partitionKey=1]
+    - SCAN[map2, partitionKey=K]
+```
+
+![Scan + Scan -> HashJoin](https://svgshare.com/i/tfk.svg)
+
+##### Member pruning for Scan -> Transform -> Aggregate
+```
+- AGGREGATION
+  - TRANSFORM
+    - SCAN[map, partitionKey=K]
+```
+
+![Scan + Scan -> HashJoin](https://svgshare.com/i/tTL.svg)
+
+On the picture above the basic units are processors, since the actual graph looks like `Scan -> Transform -> Aggregate`.
+So, here we can eliminate whole Member 2, since according to partition key condition only Member 1 contains required data
+and final destination point `Aggregate`. Moreover, it is a good example for further processor pruning optimization.
+
+#### Solution design details
+
+To support this behaviour, we need to extend the API.
+
+We propose to change the contract of `ProcessorMetaSupplier#get` return function so that it will be
```

Comment:
this is "Member pruning" chapter, ProcessorMetaSupplier#get should be under "Processor pruning"

Reply:
Actually, it's gone.